common/src/main/java/com/wynntils/mc/EventFactory.java

```diff
@@ -107,15 +107,19 @@
 /** Creates events from mixins and platform dependent hooks */
 public final class EventFactory {
     private static <T extends Event> T post(T event) {
```

Comment:
Haha. :-) Actually, my first attempt was to add a `boolean condition` to post. But then it turned out to have the same condition everywhere, except a few cases which had `true`. This made them not stand out, since you started kind of filtering out the last argument. 

I was also initially thinking "maybe some events should be sent only `onServer`, and others `onWorld`", but that was not a good approach either. What we really want to do is filter `onServer`; the few moments we spend in the hub or character selection screen is irrelevant -- the point is to not send events on single-player and other servers. A handful of features must explicitly be disabled on hub or char sel screen (like minimap), but they can do so themselves.

It was also very confusing, not knowing if the events where delivered or not. I think this is better. As soon as we have confirmed that we are on Wynncraft, we start sending events. Additional filtering can be done by listeners.